src/main/java/edu/njit/jerse/automation/AsheAutomation.java

```diff
@@ -0,0 +1,141 @@
+     * @param rootPath the root path of the project, used for determining the relative
+     *                 path of each Java file within the project structure
+     * @throws IOException if an I/O error occurs while accessing the directory
+     */
+    public static void iterateJavaFiles(File dir, String rootPath) throws IOException {
+        LOGGER.info("Iterating over Java files in directory: {}", dir.getAbsolutePath());
+
+        try (Stream<Path> paths = Files.walk(Paths.get(dir.getAbsolutePath()))) {
+            paths.filter(Files::isRegularFile)
+                    // include only files with .java extension
+                    .filter(path -> path.toString().endsWith(".java"))
+                    .forEach(path -> {
+                        try {
+                            processJavaFile(path.toFile(), rootPath);
+                        } catch (IOException | ExecutionException | InterruptedException | TimeoutException e) {
+                            LOGGER.error("Error processing Java file: {}", path.toFile().getAbsolutePath(), e);
+                            throw new RuntimeException("Error processing Java file: " + path.toFile().getAbsolutePath(), e);
+                        }
+                    });
+        }
+        LOGGER.info("Completed iterating over Java files in directory: {}", dir.getAbsolutePath());
+    }
+
+    /**
+     * Processes an individual Java file using the ASHE framework. This involves parsing
+     * the file to extract class and method information, and then for each public method,
+     * the ASHE 'run' method is invoked to apply minimization and error correction.
+     *
+     * @param javaFile        the Java file to be processed
+     * @param projectRootPath the root path of the project, used for determining the
+     *                        relative path of the Java file
+     * @throws IOException          if an I/O error occurs when opening or parsing the file
+     * @throws ExecutionException   if the ASHE framework encounters an error during execution
+     * @throws InterruptedException if the ASHE execution is interrupted
+     * @throws TimeoutException     if the ASHE execution takes longer than the allowed time
+     */
+    private static void processJavaFile(File javaFile, String projectRootPath)
+            throws IOException, ExecutionException, InterruptedException, TimeoutException {
+        LOGGER.info("Processing Java file: {}", javaFile.getAbsolutePath());
+        CompilationUnit cu = StaticJavaParser.parse(javaFile);
+
+        String targetFile = javaFile
+                .getAbsolutePath()
+                .substring(projectRootPath.length() + 1)
+                .replace(File.separatorChar, '/');
+
+        if (targetFile.startsWith("src/main/java/")) {
+            targetFile = targetFile.substring("src/main/java/".length());
+        }
+
+        String packageName = cu.getPackageDeclaration().map(NodeWithName::getNameAsString).orElse("");
```

Comment:
`packageName` is going to be tested with `isEmpty`.  I would be inclined to put all the computation of the package prefix string in one place (here) rather than some of it here and some of it in the rhs of the `fullClassName` declaration.